src/main/java/com/qunite/api/service/UserServiceImpl.java

```diff
@@ -32,6 +32,19 @@ public User createOne(User user) {
     }
   }
 
+  @Override
+  @Transactional
+  public User updateOne(User newUser) {
+    Optional<User> foundByUsername = userRepository.findUserByUsername(newUser.getUsername());
+    Optional<User> foundByEmail = userRepository.findUserByEmail(newUser.getUsername());
+    if ((foundByUsername.isEmpty() || foundByUsername.get().getId().equals(newUser.getId()))
+        && (foundByEmail.isEmpty() || foundByEmail.get().getId().equals(newUser.getId()))) {
```

Comment:
I guess we need some method like findByEmailOrUsername(String email, String username) in UserRepository

Reply:
and due to your checks, if user is not present(no username and email), update will create entity

Reply:
think about idea of updating user: you`re changing password, or you`re updating you email OR login. in your impl, yours dto contains (idk why but the fact is) email AND login. I suppose it`s impossible situation for user to update full credentials, so I propose doing smth like:
1) Change UserUpdateDto to contain String login. I don`t see scenairios when user will be update FULL credentials.
2) In this case, you can achieve changing user creds by using already EXISTING method rather than crating two separated ones:
public User update(User newUser) {
    return userRepository.findByEmailOrUsername(newUser.getLogin())
        .map(userRepository::save)
        .orElseThrow(() -> new UserAlreadyExistsException("bebeb"));
  }

Reply:
The user wants to change the username to alex, the username is not taken

findByUsernameOrEmail(alex, max@gmail.com) will find our max. What happened? Username taken? No, that's why I divided it into two methods.

Reply:
It will not create new entity because we have partial update in controller, so id and other fields(e.g. password) will be the same.

Reply:
In fact, this method is just create with validation appropriate for the update logic